File: mediq-backend/src/main/java/com/mediq/service/DoctorService.java
package com.mediq.service;

import com.mediq.dto.ConsultationResponse;
import com.mediq.dto.CreateConsultationRequest;

import java.util.List;

public interface DoctorService {

    ConsultationResponse saveConsultation(CreateConsultationRequest request);

    ConsultationResponse getConsultationById(Long consultationId);

    ConsultationResponse getConsultationByCode(String consultationCode);

    List<ConsultationResponse> getPatientConsultationHistory(Long patientId);

    List<ConsultationResponse> getDoctorConsultations(Long doctorId, Long campId);
}

File: mediq-frontend/src/services/doctorService.js
import api from './api';

export const doctorService = {
  saveConsultation: async (data) => api.post('/doctor/consultations', data),
  getConsultationById: async (id) => api.get(`/doctor/consultations/${id}`),
  getConsultationByCode: async (code) => api.get(`/doctor/consultations/code/${code}`),
  getPatientConsultationHistory: async (patientId) => api.get(`/doctor/consultations/patient/${patientId}`),
  getDoctorConsultations: async (campId, doctorId) =>
    api.get(`/doctor/consultations/camp/${campId}`, { params: { doctorId } }),
};

export default doctorService;
